tests/test_ext.py

```diff
@@ -1,35 +1,244 @@
 from __future__ import absolute_import, unicode_literals
 
-import unittest
+import mock
 
-from mopidy import config, ext
+import pkg_resources
 
+import pytest
 
-class ExtensionTest(unittest.TestCase):
+from mopidy import config, exceptions, ext
 
-    def setUp(self):  # noqa: N802
-        self.ext = ext.Extension()
+from tests import IsA, any_unicode
 
-    def test_dist_name_is_none(self):
-        self.assertIsNone(self.ext.dist_name)
 
-    def test_ext_name_is_none(self):
-        self.assertIsNone(self.ext.ext_name)
+class TestExtension(ext.Extension):
+    dist_name = 'Mopidy-Foobar'
+    ext_name = 'foobar'
+    version = '1.2.3'
 
-    def test_version_is_none(self):
-        self.assertIsNone(self.ext.version)
+    def get_default_config(self):
+        return '[foobar]\nenabled = true'
 
-    def test_get_default_config_raises_not_implemented(self):
-        with self.assertRaises(NotImplementedError):
-            self.ext.get_default_config()
 
-    def test_get_config_schema_returns_extension_schema(self):
-        schema = self.ext.get_config_schema()
-        self.assertIsInstance(schema['enabled'], config.Boolean)
+any_testextension = IsA(TestExtension)
 
-    def test_validate_environment_does_nothing_by_default(self):
-        self.assertIsNone(self.ext.validate_environment())
 
-    def test_setup_raises_not_implemented(self):
-        with self.assertRaises(NotImplementedError):
-            self.ext.setup(None)
+# ext.Extension
+
+@pytest.fixture
+def extension():
+    return ext.Extension()
+
+
+def test_dist_name_is_none(extension):
+    assert extension.dist_name is None
+
+
+def test_ext_name_is_none(extension):
+    assert extension.ext_name is None
+
+
+def test_version_is_none(extension):
+    assert extension.version is None
+
+
+def test_get_default_config_raises_not_implemented(extension):
+    with pytest.raises(NotImplementedError):
+        extension.get_default_config()
+
+
+def test_get_config_schema_returns_extension_schema(extension):
+    schema = extension.get_config_schema()
+    assert isinstance(schema['enabled'], config.Boolean)
+
+
+def test_validate_environment_does_nothing_by_default(extension):
+    assert extension.validate_environment() is None
+
+
+def test_setup_raises_not_implemented(extension):
+    with pytest.raises(NotImplementedError):
+        extension.setup(None)
+
+
+# ext.load_extensions
+
+@pytest.fixture
+def iter_entry_points_mock(request):
+    patcher = mock.patch('pkg_resources.iter_entry_points')
+    iter_entry_points = patcher.start()
+    iter_entry_points.return_value = []
+    request.addfinalizer(patcher.stop)
+    return iter_entry_points
```

Comment:
This can be changed to:

``` python
@pytest.yield_fixture
def iter_entry_points_mock(request):
    patcher = mock.patch('pkg_resources.iter_entry_points')
    iter_entry_points = patcher.start()
    iter_entry_points.return_value = []
    yield iter_entry_points
    patcher.stop()
```